src/index.js

```diff
@@ -0,0 +1,175 @@
+import './index.css';
+import reloadDom from './module/clear.js';
+
+const form = document.getElementById('form');
+const listItems = document.getElementById('list-items');
+const clearSelected = document.querySelector('.clear');
+
+class Task {
+  constructor(newData) {
+    this.newData = newData;
+  }
+
+  delete() {
+    const btns = document.querySelectorAll('#delete');
+
+    btns.forEach((btn) => {
+      const newData = this.getStoredData();
+
+      btn.addEventListener('click', (e) => {
+        const { id } = e.target.parentElement.dataset;
+        const indexItem = newData.findIndex((item) => item.index === +id);
+
+        if (indexItem > -1) {
+          newData.splice(indexItem, 1);
+
+          newData.forEach((item, indexItem) => {
+            item.index = indexItem;
+          });
+
+          localStorage.setItem('data', JSON.stringify(newData));
+          this.read();
+        }
+      });
+    });
+  }
+
+  getStoredData() {
+    const newData = JSON.parse(localStorage.getItem('data') || '[]');
+    this.newData = newData;
+    return newData;
+  }
+
+  Clear() {
```

Comment:
Kindly, cosider using an arrow function here to make your codes look more updated :smiley: